src/main/java/com/gradle/exportapi/dbutil/SqlHelper.java

```diff
@@ -1,35 +1,59 @@
 package com.gradle.exportapi.dbutil;
 
+import com.zaxxer.hikari.HikariDataSource;
 import org.knowm.yank.PropertiesUtils;
 import org.knowm.yank.Yank;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.Connection;
+import java.sql.SQLException;
 import java.util.Arrays;
 import java.util.Properties;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 public class SqlHelper {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SqlHelper.class);
-
-    public static final String GE_EXPORT_SCHEMA_PROPERTY_KEY = "geexport.sql.properties";
     private static final AtomicBoolean sqlQueriesLoaded = new AtomicBoolean(false);
+    private static final AtomicReference<String>  databaseProductName = new AtomicReference<>(null);
+
+    private static String getDatabaseProductName() {
+        if(databaseProductName.get() == null) {
```

Comment:
If this were a Kotlin class the [lazy delegate](https://kotlinlang.org/docs/reference/delegated-properties.html) would do the null-check & synchronization of this for you. 🎉 

So this would look something like
```kotlin
val databaseProductName : String by lazy {
    val connection = Yank.defaultConnectionPool.connection.metaData.databaseProductName
    LOGGER.info( /* same message you have now */ )
    return connection // return statement optional
}
```

Not saying you have to change anything about the project or this file, just dabbling in Kotlin myself and starting to recognize some of its benefits